client/gutenberg/extensions/likes-and-sharing/panel.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * External dependencies
+ */
+import get from 'lodash/get';
+
+/**
+ * Internal dependencies
+ */
+import { __ } from 'gutenberg/extensions/presets/jetpack/utils/i18n';
+import { Component } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+import { CheckboxControl, PanelBody } from '@wordpress/components';
+import { withDispatch, withSelect } from '@wordpress/data';
+
+class LikesAndSharingPanel extends Component {
+	render() {
+		const { areLikesEnabled, isSharingEnabled, toggleLikes, toggleSharing } = this.props;
+
+		return (
+			<PanelBody title={ __( 'Likes and Sharing' ) }>
+				<CheckboxControl
+					label={ __( 'Show likes.' ) }
+					checked={ areLikesEnabled }
+					onChange={ toggleLikes }
+				/>
+
+				<CheckboxControl
+					label={ __( 'Show sharing buttons.' ) }
+					checked={ isSharingEnabled }
+					onChange={ toggleSharing }
+				/>
+			</PanelBody>
+		);
+	}
+}
+
+// Fetch the post meta.
+const applyWithSelect = withSelect( select => {
+	const { getEditedPostAttribute } = select( 'core/editor' );
+	const meta = getEditedPostAttribute( 'meta' );
+
+	return {
+		areLikesEnabled: ! get( meta, [ 'switch_like_status' ] ), // todo site option
```

Comment:
based on your description it seemed like this should _not_ be negated here but I see that we are. is this correct?

Reply:
If `switch_like_status` is FALSE, we're not inverting the site setting (which for now we're assuming is TRUE), therefore `areLikesEnabled` is true when `switch_like_status` is false. (Later there will be more complexity here.)

If `sharing_disabled` is FALSE, we're not disabling sharing, therefore `isSharingEnabled` is true.

So both of these should be the opposite of their confusingly-named meta values.

Reply:
wouldn't hut to add a quick note here on those values and their meanings.

```js
/** whether we should flip the site-wide like status for this post */
const shouldFlipSiteLikeSetting = get( meta, [ 'switch_like_status' ] );

const isSharingDisabled = get( meta, [ 'sharing_disabled' ] );

return {
	areLikesEnabled = ! shouldFlipSiteLikeSetting,
	isSharingEnabled = ! isSharingDisabled,
	meta
}
```